lib/perl/Genome/Model/Tools/Vcf/FinalPassOnly.pm

```diff
@@ -0,0 +1,72 @@
+package Genome::Model::Tools::Vcf::FinalPassOnly;
+
+use Genome;
+use strict;
+use warnings;
+use Genome::File::Vcf::Reader;
+use Genome::File::Vcf::Writer;
+
+class Genome::Model::Tools::Vcf::FinalPassOnly {
+    is => 'Command',
+    has => [
+        output_file => {
+            is  => 'Text',
+            doc => "output vcf file, it can be either .vcf or .vcf.gz",
+        },
+        input_file => {
+            is  => 'Text',
+            doc => "input vcf file, it can be either .vcf or .vcf.gz",
+        },
+        sample_name => {
+            is  => 'Text',
+            doc => "Sample name, usually it is tumor sample name",
+        },
+        tabix_index => {
+            is  => 'Boolean',
+            doc => 'create tabix index for output vcf file',
+            is_optional => 1,
+            default     => 1,
+        },
+    ],
+};
+
+
+sub help_detail {
+    <<'HELP';
+   Make a new vcf file that only contains final passed variants
+
+HELP
+}
+
+
+sub execute {
+    my $self = shift;
+    my $out_file = $self->output_file;
+
+    my $in_vcf  = Genome::File::Vcf::Reader->new($self->input_file);
+    my $header  = $in_vcf->header;
+    my $out_vcf = Genome::File::Vcf::Writer->new($out_file, $header);
+
+    my $index = $header->index_for_sample_name($self->sample_name);
+
+    while (my $entry = $in_vcf->next) {
+        next if $entry->is_filtered;
+        my $filters = $entry->filters;
+        if ($entry->sample_field($index, 'FT') eq 'PASS' or $filters ~~ ['PASS']) {
+            $out_vcf->write($entry);
+        }
+    }
+    $out_vcf->close;
+
+    if ($self->tabix_index) {
+        my $cmd = Genome::Model::Tools::Tabix::Index->create(
```

Comment:
The docs for output file say it can be either `.vcf` or `.vcf.gz`, but indexing requires a compressed file.  Would it be possible to either skip this (even if `$self->tabix_index` is true) when `not Genome::Sys->file_is_gzipped($out_file)` or provide an error message about the issue when the command starts?